### PR TITLE
7.10 compatibility

### PR DESCRIPTION
**Pull request by thumphries**
- Rename cabal files with ambiata- prefix
- Bump p submodule to 7.10-compatible commit
- Fix a couple of AMP warnings
- Add submodule for htoml, pinned to `htoml-0.1.0.3`. Hackage version is stale and doesn't build

Library builds in 7.8.4 and 7.10.2

This allows the merge of ambiata/mafia#23 and upgrade of anything depending on x

### CODE REVIEW

**Comment by thumphries** [comment]
Are we OK with an external submodule? Should we fork, or use Huw's fork instead? htoml package creator is MIA

**Comment by charleso** [comment]
> Are we OK with an external submodule?

Not really. Be good to fork to `ambiata` if possible.

File: x-aeson/src/X/Data/Aeson.hs [hunk 1]
```diff
@@ -19,6 +19,7 @@ import           Data.Text
 import           Data.Text.Encoding (decodeUtf8, encodeUtf8)
 import           Data.ByteString.Lazy (toStrict)
 
+import           Prelude -- AMP compatibility workaround
```
Comment (charleso) [inline x-aeson/src/X/Data/Aeson.hs]: Dumb/lazy question - what happens without this?

Comment (thumphries) [inline x-aeson/src/X/Data/Aeson.hs]: Redundant Applicative warnings. This is the usual migration trick, but given we maintain our own Prelude, maybe not so good. Can be replaced with `NoImplicitPrelude` and explicit imports `Control.Monad` `Data.Either`, `Data.Maybe`, `GHC.Base`

Comment (thumphries) [inline x-aeson/src/X/Data/Aeson.hs]: Other packages can get away with plain `NoImplicitPrelude` ; `import P` afaik

Comment (jystic) [inline x-aeson/src/X/Data/Aeson.hs]: [Explicit imports](https://github.com/ambiata/guideline/blob/master/language/haskell.md#imports) sound like a better solution to me if we can get that to work

Comment (thumphries) [inline x-aeson/src/X/Data/Aeson.hs]: Yeah, that seems more future-proof, given this trick kinda relies on an arcane implementation detail

**Comment by thumphries** [comment]
@charleso @jystic @tmcgilchrist this should be good to go

File: x-aeson/src/X/Data/Aeson.hs [hunk 2]
```diff
@@ -10,15 +11,18 @@ module X.Data.Aeson (
...
 import           Data.Aeson.Types as X
+import           Data.Either (Either(..), either)
+import           Data.Functor ((<$>))
+import           Data.Maybe (Maybe, maybe)
 import           Data.Monoid
 import           Data.Text
 import           Data.Text.Encoding (decodeUtf8, encodeUtf8)
 import           Data.ByteString.Lazy (toStrict)
-
+import           GHC.Base ((.), ($))
```
Comment (jystic) [inline x-aeson/src/X/Data/Aeson.hs]: It probably doesn't matter, but I think we should import these from `Data.Function` instead. The `GHC.*` modules are really internal implementation details.

File: x-aeson/test/Test/X/Data/Aeson.hs [hunk 3]
```diff
@@ -1,13 +1,18 @@
...
-
+import           Data.Either
+import           Data.Functor ((<$>))
+import           Data.Maybe
 import           Data.Text as T
 
 import           Disorder.Aeson
 
+import           GHC.Base
+import           GHC.Show
```
Comment (jystic) [inline x-aeson/test/Test/X/Data/Aeson.hs]: Not sure what these imports are for but you can get the `Show` type class in `Text.Show`

Comment (thumphries) [inline x-aeson/test/Test/X/Data/Aeson.hs]: yep done

**Comment by jystic** [comment]
Looks good, nice work! :8ball:

**Pull request merged**